### ISSUE TITLE
Android: TypeError: Cannot read property 'Symbol(Symbol.toStringTag)' of undefined

### ISSUE THREAD

**Issue by dukewan** [nativescript-community/ui-canvas#28]
### Make sure to check the demo app(s) for sample usage 
Done.

### Make sure to check the existing issues in this repository 
Done.

### If the demo apps cannot help and there is no issue for your problem, tell us about it
Please, ensure your title is less than 63 characters long and starts with a capital
letter.

### Which platform(s) does your issue occur on?
- Android
- 7
- device

### Please, provide the following version numbers that your issue occurs with:

- CLI: 7.1.2-2021-01-14-110216-14335
- Cross-platform modules: 7.2.0
- Runtime(s): 7.0.1
- Plugin(s):

```
"dependencies": {
    "@knotes/nativescript-camera-plus": "^4.0.0",
    "@knotes/nativescript-clipboard": "^2.1.0",
    "@knotes/nativescript-cscreenshot": "^1.0.0",
    "@knotes/nativescript-generate-pdf": "^1.0.2",
    "@knotes/nativescript-keyboard-toolbar": "^2.0.0",
    "@knotes/nativescript-share-file": "^1.0.1",
    "@nativescript-community/fonticon": "^3.0.0",
    "@nativescript-community/ui-canvas": "^4.0.40",
    "@nativescript-community/ui-image": "^4.0.19",
    "@nativescript/background-http": "^5.0.1",
    "@nativescript/core": "^7.2.0",
    "@nativescript/firebase": "^11.1.3",
    "@nativescript/imagepicker": "^1.0.2",
    "@nativescript/iqkeyboardmanager": "^2.0.0",
    "@nativescript/local-notifications": "^5.1.0",
    "@nativescript/social-share": "^2.0.1",
    "@nativescript/theme": "^3.0.0",
    "@nstudio/nativescript-cardview": "^2.0.1",
    "@nstudio/nativescript-checkbox": "^2.0.4",
    "@nstudio/nativescript-loading-indicator": "^4.0.0",
    "@proplugins/nativescript-orientation": "^2.5.0",
    "@proplugins/nativescript-purchase": "^6.0.0",
    "@triniwiz/nativescript-toasty": "^4.0.4",
    "@vue/devtools": "^5.3.3",
    "axios": "^0.19.2",
    "bson-objectid": "^1.3.1",
    "crypto-js": "^4.0.0",
    "handlebars": "^4.7.6",
    "lodash": "^4.17.15",
    "marked": "^1.2.2",
    "md5": "^2.3.0",
    "moment": "^2.27.0",
    "nativescript-apple-sign-in": "^2.0.0",
    "nativescript-cscreenshot": "^1.0.2",
    "nativescript-inappbrowser": "^3.0.1",
    "nativescript-socketio": "^3.3.1",
    "nativescript-sqlite": "^2.6.6",
    "nativescript-taptic-engine": "^2.1.0",
    "nativescript-toasty": "^3.0.0-alpha.2",
    "nativescript-ui-listview": "^9.1.0",
    "nativescript-vue": "^2.8.1",
    "nativescript-vue-devtools": "^1.4.0",
    "nativescript-windowed-modal": "^7.0.0",
    "typescript": "^4.1.3",
    "vuex": "^3.5.1",
    "vuex-i18n": "^1.13.1"
  },
  "devDependencies": {
    "@babel/core": "^7.11.6",
    "@babel/preset-env": "^7.11.5",
    "@nativescript/android": "7.0.1",
    "@nativescript/webpack": "^3.0.4",
    "babel-loader": "^8.1.0",
    "handlebars-loader": "^1.7.1",
    "nativescript-vue-template-compiler": "^2.8.1",
    "nativescript-worker-loader": "~0.12.1",
    "node-sass": "^4.14.1",
    "sass": "^1.26.10",
    "vue-loader": "^15.9.3"
  }
```

### Please, tell us how to recreate the issue in as much detail as possible. 
Describe the steps to reproduce it.

The error `TypeError: Cannot read property 'Symbol(Symbol.toStringTag)' of undefined` arises whenever I use `CanvasView` on android device. It happens while initialization.

More details:
![8831609120445_ pic_hd](https://user-images.githubusercontent.com/2100515/103184681-ff7dc600-486d-11eb-9337-e6e7d93ca259.jpg)

### Is there any code involved? 

I use nativescript-vue:

```
import CanvasPlugin from '@nativescript-community/ui-canvas/vue'
Vue.use(CanvasPlugin)
```

Basically when CanvasView is used the error arises so it errors out when initializing.

```
<CanvasView
  id="canvasTarget"
  @draw="canvasTargetReady"
  left="0" top="0" width="100%" height="100%" />
<CanvasView
  id="canvas"
  @draw="canvasReady" @touch="onCanvasTouch"
  left="0" top="0" width="100%" height="100%"
  />
```

**Comment by farfromrefug** [comment]
@dukewan i have some issues understanding here. 
your package.json mentions the community version of the plugin (the one you should use). but you mention `nativescript-canvas` as import. this would not work.
also there are 2 canvasview in your sample code. please could you try with one for now. and what s in your canvasTargetReady?
seems to me like you are trying to access the canvas when it does not exist.

**Comment by dukewan** [comment]
Sorry, I copied the description from my previous [issue](https://github.com/nativescript-community/ui-canvas/issues/27) which is NS6. But the NS7 version did throw this error after migration.

I also tried to comment out one of  the CanvasView in my code and it still throws this error. I tried to comment both of them and no error throwed.

I think the error exists when CanvasView is initialized on page load, because I tried to comment out the event handlers and it did not help at all.

**Comment by farfromrefug** [comment]
@dukewan I don't think this is the issue. I have 5 apps heavily using it. 
I can fix it if you can create a small  repro example (no other plugin)

**Comment by dukewan** [comment]
This is the repro demo: https://github.com/dukewan/canvas-repro
Steps:
1. `cd demo`
2. `ns debug android`
3. tap the label to test canvas

While I was testing it, if two canvasview are used, the error always throws. If I comment out one of them, the error throws when entering the demo page for the second time.

**Comment by farfromrefug** [comment]
@dukewan seeing your code explains it all :D You cant use the canvas object outside of the `draw` method!
This an android limitation.
If you want to do something on touch, then call invalidate and handle the change in the `draw` method.

**Comment by dukewan** [comment]
Sorry for the late reply, I don't think it's becuase of using the cavas object in the touch handler. I removed the touch handler and the error still exists: https://github.com/dukewan/canvas-repro/blob/main/demo/src/components/Test.vue

Also if we can't use the canvas object outside draw method, how do we call invalidate? As far as I know, this method only exists in canvas object.

I believe the error arised in the draw method of the second canvasview while initializing.

**Comment by farfromrefug** [comment]
@dukewan there are 2 levels. the CanvasView (the one in your layout) on which you can call invalidate.
And there is Canvas which you get from the draw event. 
it is Canvas you CANT store nore use outside the draw event. its existence is bound to the scope of the draw method. 
so touch => call invalidate on your CanvasView. in the next draw event draw on the Canvas with the touch data you for before

**Comment by MattCCC** [comment]
I played a bit with it some time ago and the issue is same if you try to call CanvasVue within a Vue component that is loaded to a page. I believe it's easily reproducible on latest ns vue + ns 7.3.0

**Comment by farfromrefug** [comment]
@MattCCC i use Canvas a lot with Vue and have no issue. We need a reproducable sample to see what s wrong

**Comment by dukewan** [comment]
Agree with @MattCCC , the issue is describe clearly as "the issue is same if you try to call CanvasVue within a Vue component that is loaded to a page"

**Comment by Upd4ting** [comment]
This issue got a fix? I'm having the same issue in ui-chart plugins

**Comment by farfromrefug** [comment]
@Upd4ting please share how you use the canvas plugin

**Comment by Upd4ting** [comment]
@farfromrefug using ui-chart plugin

Registering it in my main.ts file
```
import { install } from '@nativescript-community/ui-chart';
install();
Vue.registerElement('LineChart', () => require('@nativescript-community/ui-chart/charts').LineChart);
```

Then just placing that in my page

`<LineChart ref="chart" width="100%" height="100%" backgroundColor="transparent" @loaded="onChartLoaded" />`

That crash with `"TypeError: Cannot read property 'Symbol(Symbol.toStringTag)' of undefined"` at rendering time

**Comment by farfromrefug** [comment]
@Upd4ting ok so different! I need the full error with callstack

**Comment by Upd4ting** [comment]
```
[Vue warn]: Error in v-on handler: "TypeError: Cannot read property 'Symbol(Symbol.toStringTag)' of undefined"
JS:
JS: found in
JS:
JS: ---> <Stats> at app/pages/Stats.vue
JS:        <Frame>
JS:          <Root>
JS: discardedErrorEvent
JS: Error: Calling js method onCreateView failed
JS: TypeError: Cannot read property 'Symbol(Symbol.toStringTag)' of undefined
```
thats the only thing i got

**Comment by farfromrefug** [comment]
@Upd4ting you dont have more? I would need to know where it actually fails.
Another solutoin is for you to recreate it in a sample app with a chart. Could you try?

**Comment by Upd4ting** [comment]
@farfromrefug 
is that better ?

```
System.err: An uncaught Exception occurred on "main" thread.
System.err: Calling js method onCreateView failed
System.err: TypeError: Cannot read property 'Symbol(Symbol.toStringTag)' of undefined
System.err:
System.err: StackTrace:
System.err: get(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript-community\ui-canvas\canvas.android.js:121:0)
System.err:     at isPlainObject$1(file: app\webpack:\harmopool-mobile-app\node_modules\nativescript-vue\dist\index.js:76:0)
System.err:     at observe(file: app\webpack:\harmopool-mobile-app\node_modules\nativescript-vue\dist\index.js:992:0)
System.err:     at defineReactive(file: app\webpack:\harmopool-mobile-app\node_modules\nativescript-vue\dist\index.js:1028:0)
System.err:     at walk(file: app\webpack:\harmopool-mobile-app\node_modules\nativescript-vue\dist\index.js:940:0)
System.err:     at Observer(file: app\webpack:\harmopool-mobile-app\node_modules\nativescript-vue\dist\index.js:928:0)
System.err:     at observe(file: app\webpack:\harmopool-mobile-app\node_modules\nativescript-vue\dist\index.js:996:0)
System.err:     at reactiveSetter(file: app\webpack:\harmopool-mobile-app\node_modules\nativescript-vue\dist\index.js:1062:0)
System.err:     at proxySetter(file: app\webpack:\harmopool-mobile-app\node_modules\nativescript-vue\dist\index.js:2857:0)
System.err:     at onChartLoaded(file: app\webpack:\harmopool-mobile-app\app\pages\Stats.vue:238:0)
System.err:     at invokeWithErrorHandling(file: app\webpack:\harmopool-mobile-app\node_modules\nativescript-vue\dist\index.js:1862:0)
System.err:     at invoker(file: app\webpack:\harmopool-mobile-app\node_modules\nativescript-vue\dist\index.js:2519:0)
System.err:     at _handleEvent(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\data\observable\index.js:233:0)
System.err:     at notify(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\data\observable\index.js:216:0)
System.err:     at _emit(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\data\observable\index.js:256:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:240:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view\index.android.js:309:0)
System.err:     at (file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:318:0)
System.err:     at callFunctionWithSuper(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:312:0)
System.err:     at callLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:318:0)
System.err:     at loadView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:458:0)
System.err:     at (file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:237:0)
System.err:     at eachChildView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\layouts\layout-base-common.js:101:0)
System.err:     at eachChild(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view\view-common.js:759:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:236:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view\index.android.js:309:0)
System.err:     at (file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:318:0)
System.err:     at callFunctionWithSuper(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:312:0)
System.err:     at callLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:318:0)
System.err:     at loadView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:458:0)
System.err:     at (file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:237:0)
System.err:     at eachChildView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\layouts\layout-base-common.js:101:0)
System.err:     at eachChild(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view\view-common.js:759:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:236:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view\index.android.js:309:0)
System.err:     at (file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:318:0)
System.err:     at callFunctionWithSuper(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:312:0)
System.err:     at callLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:318:0)
System.err:     at loadView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:458:0)
System.err:     at (file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:237:0)
System.err:     at eachChildView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\layouts\layout-base-common.js:101:0)
System.err:     at eachChild(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view\view-common.js:759:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:236:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view\index.android.js:309:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\nativescript-carousel\carousel.common.js:41:0)
System.err:     at (file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:318:0)
System.err:     at callFunctionWithSuper(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:312:0)
System.err:     at callLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:318:0)
System.err:     at loadView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:458:0)
System.err:     at (file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:237:0)
System.err:     at eachChildView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\layouts\layout-base-common.js:101:0)
System.err:     at eachChild(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view\view-common.js:759:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:236:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view\index.android.js:309:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\nativescript-carousel\carousel.android.js:135:0)
System.err:     at (file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:318:0)
System.err:     at callFunctionWithSuper(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:312:0)
System.err:     at callLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:318:0)
System.err:     at loadView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:458:0)
System.err:     at (file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:237:0)
System.err:     at eachChildView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\layouts\layout-base-common.js:101:0)
System.err:     at eachChild(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view\view-common.js:759:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:236:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view\index.android.js:309:0)
System.err:     at (file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:318:0)
System.err:     at callFunctionWithSuper(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:312:0)
System.err:     at callLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:318:0)
System.err:     at loadView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:458:0)
System.err:     at (file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:237:0)
System.err:     at eachChildView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\content-view\index.js:65:0)
System.err:     at eachChildView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\page\page-common.js:106:0)
System.err:     at eachChild(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view\view-common.js:759:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:236:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view\index.android.js:309:0)
System.err:     at onLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\page\index.android.js:38:0)
System.err:     at (file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:318:0)
System.err:     at callFunctionWithSuper(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:312:0)
System.err:     at callLoaded(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:318:0)
System.err:     at loadView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:458:0)
System.err:     at _addViewCore(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:453:0)
System.err:     at _addView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\core\view-base\index.js:439:0)
System.err:     at onCreateView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\frame\index.android.js:730:0)
System.err:     at onCreateView(file: app\webpack:\harmopool-mobile-app\node_modules\@nativescript\core\ui\frame\fragment.android.js:25:0)
System.err:     at com.tns.Runtime.callJSMethodNative(Native Method)
System.err:     at com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:1302)
System.err:     at com.tns.Runtime.callJSMethodImpl(Runtime.java:1188)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:1175)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:1153)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:1149)
System.err:     at com.tns.FragmentClass.onCreateView(FragmentClass.java:55)
System.err:     at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2699)
System.err:     at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:320)
System.err:     at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1199)
System.err:     at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1368)
System.err:     at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1446)
System.err:     at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1509)
System.err:     at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:447)
System.err:     at androidx.fragment.app.FragmentManager.executeOps(FragmentManager.java:2181)
System.err:     at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2004)
System.err:     at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1959)
System.err:     at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1861)
System.err:     at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:413)
System.err:     at android.os.Handler.handleCallback(Handler.java:938)
System.err:     at android.os.Handler.dispatchMessage(Handler.java:99)
System.err:     at android.os.Looper.loop(Looper.java:223)
System.err:     at android.app.ActivityThread.main(ActivityThread.java:7656)
System.err:     at java.lang.reflect.Method.invoke(Native Method)
System.err:     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
JS: uncaughtErrorEvent
JS: Error: Calling js method onCreateView failed
JS: TypeError: Cannot read property 'Symbol(Symbol.toStringTag)' of undefined
System.err: com.tns.NativeScriptException: Error: Calling js method onCreateView failed
System.err: TypeError: Cannot read property 'Symbol(Symbol.toStringTag)' of undefined
System.err:     at com.tns.Runtime.passExceptionToJsNative(Native Method)
System.err:     at com.tns.Runtime.passUncaughtExceptionToJs(Runtime.java:84)
System.err:     at com.tns.NativeScriptUncaughtExceptionHandler.uncaughtException(NativeScriptUncaughtExceptionHandler.java:34)
System.err:     at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1073)
System.err:     at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1068)
System.err:     at java.lang.Thread.dispatchUncaughtException(Thread.java:2203)
```

**Comment by farfromrefug** [comment]
@Upd4ting much better!!!! what s your version of the chart plugin and the canvas plugin? (look in node_modules and the package.json files)

**Comment by Upd4ting** [comment]
ui-chart 1.1.42
not having ui-canvas in my package.json as its installed in depency of ui-chart

**Comment by farfromrefug** [comment]
but you have it in node_modules. PLease give me the version

**Comment by Upd4ting** [comment]
4.1.2

**Comment by farfromrefug** [comment]
@Upd4ting please add the canvas plugin to your package.json and ensure you have version `4.1.3`.  Let me know if it fixes the issue

**Comment by Upd4ting** [comment]
@farfromrefug seems to work, thanks!

**Comment by farfromrefug** [comment]
@Upd4ting awesome!
@dukewan i ll close this issue for now. tell me if you still see the issue
